pkg/api/types.go

```diff
@@ -925,6 +925,15 @@ func (p *Properties) IsHostedMasterProfile() bool {
 	return p.HostedMasterProfile != nil
 }
 
+// IsUserAssignedIdentityEnabled returns true if the cluster uses managed identity
```

Comment:
We actually already have a method called `UserAssignedIDEnabled` (part of the `KubernetesConfig` type spec) so we don't need this new one.

Reply:
I've updated to use `UserAssignedIDEnabled` and I feel it's a little bit awkward: I have to judge whether `OrchestratorProfile` and `KubernetesConfig` is `nil` every time before I call this method.